dolphinscheduler-service/src/main/java/org/apache/dolphinscheduler/service/permission/ResourcePermissionCheckServiceImpl.java

```diff
@@ -111,17 +116,375 @@ public List<AuthorizationType> authorizationTypes() {
 
         @Override
         public boolean permissionCheck(int userId, String url, Logger logger) {
-            // all users can create projects
             return true;
         }
-
         @Override
         public Set<Integer> listAuthorizedResource(int userId, Logger logger) {
             return projectMapper.listAuthorizedProjects(userId, null).stream().map(Project::getId).collect(toSet());
         }
     }
 
 
+    @Component
+    public static class K8sNamespaceResourceList implements ResourceAcquisitionAndPermissionCheck<Integer> {
+
+        private final K8sNamespaceMapper k8sNamespaceMapper;
+
+       
+
+        public K8sNamespaceResourceList(K8sNamespaceMapper k8sNamespaceMapper) {
+            this.k8sNamespaceMapper = k8sNamespaceMapper;
+        }
+
+        @Override
+        public List<AuthorizationType> authorizationTypes() {
+            return Collections.singletonList(AuthorizationType.K8S_NAMESPACE);
+        }
+
+        @Override
+        public boolean permissionCheck(int userId, String url, Logger logger) {
+           return true;
+        }
+
+        @Override
+        public Set<Integer> listAuthorizedResource(int userId, Logger logger) {
+            return Collections.emptySet();
+        }
+    }
+
+
+    @Component
+    public static class EnvironmentResourceList implements ResourceAcquisitionAndPermissionCheck<Integer> {
+
+        private final EnvironmentMapper environmentMapper;
+
+       
+
+        public EnvironmentResourceList(EnvironmentMapper environmentMapper) {
+            this.environmentMapper = environmentMapper;
+        }
+
+        @Override
+        public List<AuthorizationType> authorizationTypes() {
+            return Collections.singletonList(AuthorizationType.ENVIRONMENT);
+        }
+
+        @Override
+        public boolean permissionCheck(int userId, String url, Logger logger) {
+           return true;
+        }
+
+        @Override
+        public Set<Integer> listAuthorizedResource(int userId, Logger logger) {
+            return Collections.emptySet();
```

Comment:
The data source belongs to the resource permission and cannot return an empty collection.